### PR TITLE
update _samtools and fix a bug

### PR DESCRIPTION
**Pull request by cgjosephlee**
Update to samtools v1.8, but not all subcommands are implemented.

And v1.9 removes some options in mpileup, I haven't updated to this version yet.

### CODE REVIEW

**Comment by cgjosephlee** [comment]
Just curious, how does this `if` clause work?
https://github.com/kloetzl/biozsh/blob/c163056182bd1fcceefae3516695abdaba37eac4/completions/_samtools#L271

I guess if the subcommand is not defined, it will show a warning message. I have tried a few circumstances and did not see the message popup.
Would you please give a brief explanation?

**Comment by kloetzl** [comment]
Thanks for your work!

> Would you please give a brief explanation?

Quoting from the [docs](http://zsh.sourceforge.net/Doc/Release/Completion-System.html):

> If a function name exists, it is called with the arguments args. The return argument gives the name of a parameter in which the return status from the function name should be stored; if return is empty or a single hyphen it is ignored.
>
> The return status of _call_function itself is zero if the function name exists and was called and non-zero otherwise.

If you do not see the message, maybe your zsh theme simply just doesn't show them? I had to add the following lines to my dotfiles to see them. [source](https://github.com/kloetzl/dotfiles/blob/master/zsh/completion.zsh#L90-L95).

```zsh
# Display message when no matches are found
# formatting and messages
zstyle ':completion:*' verbose yes
zstyle ':completion:*:descriptions' format '%B%d%b'
zstyle ':completion:*:messages' format '%d'
zstyle ':completion:*:warnings' format 'No matches for: %B%d%b'
```

File: completions/_samtools [hunk 1]
```diff
@@ -162,7 +162,12 @@ _samtools-sort() {
 		'-l+[compression level, from 0 to 9 (default: -1)]:int:'
 		'-m+[max memory per thread; suffix K/M/G recognize (default: 768M)]:int:'
 		'-n[sort by read name]'
-		'-o[final output to stdout]'
+		'-t[sort by value of TAG, uses position as secondary index (or read name if -n is set)]:str:'
+		'-o+[write final output to FILE]:files:_files'
+		'-O+[output format]:str:((
+			SAM\:SAM
```
Comment (kloetzl) [inline completions/_samtools]: When the description is the same as the key a simpler format can be used. `-O+[output format]:str:(SAM BAM CRAM)`. See [the docs](http://zsh.sourceforge.net/Doc/Release/Completion-System.html) under `_arguments`.

File: completions/_samtools [hunk 2]
```diff
@@ -102,7 +102,7 @@ _samtools-merge() {
 		'-1[compress level 1]'
 		'-@[number of BAM compression thread]:int:'
 		'-f[overwrite the output BAM if exist]'
-		'-h[copy the header in FILE to <out.bam>:FILE:_files'
+		'-h[copy the header in FILE to <out.bam>]:FILE:_files'
```
Comment (kloetzl) [inline completions/_samtools:105]: Good catch.

File: completions/_samtools [hunk 3]
```diff
@@ -162,7 +162,12 @@ _samtools-sort() {
 		'-l+[compression level, from 0 to 9 (default: -1)]:int:'
 		'-m+[max memory per thread; suffix K/M/G recognize (default: 768M)]:int:'
 		'-n[sort by read name]'
-		'-o[final output to stdout]'
+		'-t[sort by value of TAG, uses position as secondary index (or read name if -n is set)]:str:'
+		'-o+[write final output to FILE]:files:_files'
```
Comment (kloetzl) [inline completions/_samtools:166]: If the output is in SAM, BAM, or CRAM format, maybe we can restrict suggestions to `*.sam` etc. here?

**Review by kloetzl — CHANGES_REQUESTED** [review]

File: completions/_samtools (hunk 1 above)
Comment (cgjosephlee) [reply completions/_samtools]: Thank you. I will tidy this up.

File: completions/_samtools (hunk 3 above)
Comment (cgjosephlee) [reply completions/_samtools:166]: Well, sometimes I use completion for naming prefix therefore I would not second this. But this is good for input suggestions.
e.g. auto-complete `abc123def456.fastq.gz` and thereafter edit as `abc123def456.sam`

Comment (kloetzl) [reply completions/_samtools:166]: You are right, output is different from input because the target file may not exist yet.

**Comment by cgjosephlee** [comment]
update `samtools view` options.

**Comment by cgjosephlee** [comment]
Commits rebased!

**Comment by kloetzl** [comment]
Merged, thanks!

**Pull request closed without merging**